pkg/network/proxy.go

```diff
@@ -1,15 +1,36 @@
 package network
 
 import (
+	"log"
 	"net"
+	"os"
+	"path/filepath"
 	"time"
 
 	"github.com/pkg/errors"
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	operv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-network-operator/pkg/render"
 	k8sutil "github.com/openshift/cluster-network-operator/pkg/util/k8s"
 )
 
+// ShouldDeployKubeProxy determines if the desired network type should
+// install kube-proxy.
+// openshift-sdn and OVN-Kubernetes deploy their own kube-proxy. All other
+// network providers are assumed to require kube-proxy
+func ShouldDeployKubeProxy(conf *operv1.NetworkSpec) bool {
+	switch conf.DefaultNetwork.Type {
+	case operv1.NetworkTypeOpenShiftSDN:
+		return false
+	case operv1.NetworkTypeOVNKubernetes:
+		return false
+	default:
+		log.Printf("NOTICE: Unknown network type %s, installing kube-proxy", conf.DefaultNetwork.Type)
```

Comment:
this will end up being logged repeatedly, since this function is called from multiple places

Reply:
yeah, it's probably a bit too verbose. removed.